httpx/_concurrency.py

```diff
@@ -0,0 +1,31 @@
+import typing
```

Comment:
Let me know if this module and/or type checking stuff is over-the-top 🙃. The type checking does allow us to assert that `asyncio.Event` and `trio.Event` expose similar-enough interfaces.

Reply:
I’d probably not bother with the Protocol level stuff here yeah.

Reply:
OK I've reduced the type-checking stuff and got rid of this new `_concurrency` module :+1: